File: app/src/main/java/org/gwnu/tutorial/retrofit/RetrofitClient.java
package org.gwnu.tutorial.retrofit;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    private static RetrofitClient ourInstance = new RetrofitClient();
    public static RetrofitClient getInstance() {
        return ourInstance;
    }
    private RetrofitClient() {
    }

    Retrofit retrofit = new Retrofit.Builder()
            .baseUrl("https://api.github.com/")
            .addConverterFactory(GsonConverterFactory.create()) // 파싱등록
            .build();

    RetrofitService service = retrofit.create(RetrofitService.class);

    public RetrofitService getService() {
        return service;
    }
}

File: app/src/main/java/org/gwnu/tutorial/retrofit2/RetrofitClient.kt
package org.gwnu.tutorial.retrofit2

import android.util.Log
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

// 코틀린에서 object: 싱글턴
object RetrofitClient {
    private val TAG = RetrofitClient::class.java.simpleName

    //레트로핏 클라이언트 선언
    private var retrofitClient: Retrofit? = null
//    private lateinit var retrofitClient: Retrofit

    //레트로핏 클라이언트 가져오기
    fun getClient(baseUrl: String): Retrofit? {
        Log.d(TAG, "RetrofitClient - getClient() called ")

        if (retrofitClient == null) {

            //레트로핏 빌더를 통해 인스턴스 생성
            retrofitClient = Retrofit.Builder()
                    .baseUrl(baseUrl)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
        }
        return retrofitClient
    }

}